lib/pg_eventstore/query_builders/events_filtering_query.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+module PgEventstore
+  module QueryBuilders
+    class EventsFiltering
```

Comment:
I think we can split this class  base on all or regular stream
I would keep the same naming in both classes like `add_stream_direction` but it would produce of course different result for the all than particular stream. Let's discuss it.

Reply:
Currently this implementation doesn't know anything about "all"/"specific" stream definition. It just builds an SQL to query the database. I don't see the point of performing such split.